sentry_streams/sentry_streams/pipeline/pipeline.py

```diff
@@ -175,9 +198,9 @@ class StreamSource(Source):
     header_filter: Optional[Tuple[str, bytes]] = None
     step_type: StepType = StepType.SOURCE
 
-    def __post_init__(self) -> None:
-        super().__post_init__()
-        self.ctx.register_source(self)
+    def register(self, ctx: Pipeline, previous: Optional[Step] = None) -> None:
+        super().register(ctx)
+        ctx.register_source(self)
```

Comment:
It seems to me the only reason we need this method is so that we can call `ctx.register_source(self)`. 
This is tech debt from before the chain api when steps were defined individually so they needed a reference to the pipeline to register themselves as part of the pipeline. 

This should not be needed anymore with the chain api as steps by being passed to the `apply` method of the pipeline object. Which means the pipeline object can register the steps as part of the graph and the steps itself does not need to know about the pipeline.

That would simplify the references a lot. Do you plan to remove this indirection or is it here to stay ?  I'd try to remove this method entirely and do everything needed inside the methods of the `Pipeline` object.